fastapi/encoders.py

```diff
@@ -312,11 +298,9 @@ def jsonable_encoder(
             )
         return encoded_list
 
-    if type(obj) in ENCODERS_BY_TYPE:
-        return ENCODERS_BY_TYPE[type(obj)](obj)
-    for encoder, classes_tuple in encoders_by_class_tuples.items():
-        if isinstance(obj, classes_tuple):
-            return encoder(obj)
+    for base in obj.__class__.__mro__[:-1]:
+        if base in ENCODERS_BY_TYPE:
+            return ENCODERS_BY_TYPE[base](obj)
```

Comment:
Are you sure that `ENCODERS_BY_TYPE` always include the root class of the inheritance tree, and cannot include more nested classes?

Reply:
I'm not sure to understand, what problem do you have in mind? Can you give an example?

Reply:
Sorry, my bad

Reply:
No problem! Better safe than sorry. :)